packages/components/src/box-control/unit-control.js

```diff
@@ -35,7 +35,6 @@ export default function BoxUnitControl( {
 				<UnitControl
 					aria-label={ label }
 					className="component-box-control__unit-control"
-					hideHTMLArrows
```

Comment:
In BoxControl, spin buttons were always hidden, even when units were disabled. (Units get disabled in the "linked" control when the units of the separate box sides are "mixed".)

However, other BoxControl-like controls in the block editor like the [Spacing](https://github.com/WordPress/gutenberg/blob/ca37b82de4700065d5fedbe8829768144e8bfaeb/packages/block-editor/src/components/spacing-sizes-control/spacing-input-control.js#L219) and [Border Radius](https://github.com/WordPress/gutenberg/blob/ca37b82de4700065d5fedbe8829768144e8bfaeb/packages/block-editor/src/components/border-radius-control/all-input-control.js#L58) controls use a raw UnitControl and do not hide spin buttons when units are disabled.

@jasmussen Any opinions on whether we should unify these hiding behaviors in the UnitControls? The options are:

1. In a UnitControl, always show spin buttons when the units are disabled.
2. In a UnitControl, always hide spin buttons, regardless of whether units are disabled.
3. The consumers of UnitControl should be able to choose whether spin button are hidden. (In this case, we'll also want to iron out the inconsistencies in the BoxControl-like controls I described above.)

Reply:
Just to be sure I understand you right, when you refer to spin buttons, you refer to these?

<img width="43" alt="image" src="https://user-images.githubusercontent.com/1204802/189156530-9f6c986e-aafb-4a33-b00d-2d4c7fca77c0.png">

If yes, then I haven't seen a case where we need to/should surface these in the editor. We have a click & drag behavior:
![click and drag](https://user-images.githubusercontent.com/1204802/189156881-100b3856-0766-442e-b7a4-38bbcc107c68.gif)

There's also up arrow for incrementing, down arrow for decrementing, and shift up/down for doing that in increments of ten. Those may be less discoverable, but feel far more ergonomic to use than the tiny little hit areas of the up/down arrows. What do you think?

Reply:
IMO, in general showing spin buttons would be a good idea and would improve the input's usability. Spin buttons would offer a very "visually explicit" alternative to increase/decrease the value in the input field (while every other alternative mentioned is more implicit and hard to discover).


> Any opinions on whether we should unify these hiding behaviors in the UnitControls? The options are:
> 
> 1. In a UnitControl, always show spin buttons when the units are disabled.
> 2. In a UnitControl, always hide spin buttons, regardless of whether units are disabled.
> 3. The consumers of UnitControl should be able to choose whether spin button are hidden. (In this case, we'll also want to iron out the inconsistencies in the BoxControl-like controls I described above.)

I would vote for 1 or 3.

> Units get disabled in the "linked" control when the units of the separate box sides are "mixed"

Just to understand — in that case, `BoxControl` seems to show the word "Mixed", and so I'm not sure how spin buttons would be useful in this scenario?

<img width="272" alt="Screenshot 2022-09-09 at 18 23 06" src="https://user-images.githubusercontent.com/1083581/189396562-4e15754a-deb0-4a11-998f-eda2a8eec772.png">

Reply:
Given how small the buttons are, and the fact that you can at any one point type in an explicit value, the spin buttons really don't seem worth optimizing for.

Reply:
My original intent was to discuss step buttons in UnitControl only, but I guess now we're discussing whether to nix step buttons in general, including normal NumberControls.

I just remembered that we have a huge move in the other direction, waiting on our todo list to implement:

<img width="134" alt="NumberControl mockup with large step buttons" src="https://user-images.githubusercontent.com/555336/189757399-3f4038c5-2769-4c16-a8da-174c277aba11.png">

And IIRC, @mtias wrote somewhere that UnitControl was originally intended to be accompanied by a slider, like so:

<img width="270" alt="UnitControl with slider" src="https://user-images.githubusercontent.com/555336/189761243-8c2d1a77-2605-4c2a-95e9-dc83ef9ff668.png">

So perhaps the unifying idea is that number inputs *should* have some sort of visually obvious auxiliary control, that is *usable on a touch device*. In this case, the tiny built-in step buttons do not satisfy this requirement anyway.

tl;dr — I guess I'm fine with removing the step buttons from UnitControl right now. And they'll also be removed from NumberControl once the large custom step buttons are implemented.

Reply:
The slider and the bigger plus/minuses are both more ergonomic tools, excellent point.

Reply:
Indeed, the big + – is meant to cover incremental steps where each step is meaningful, like line height, while the slider is meant to cover wider ranges. A + – input should not be paired with a slider; custom font-size really need to be paired with the slider.

Reply:
In case where `UnitControl` will need to be adjusted with small incremental steps, would we show the big "+" / "-" buttons next to the unit dropdown?

Reply:
What's the unit dropdown? + / – should show inside the control like the line height example.

Reply:
> What's the unit dropdown?

The unit dropdown in the unit control component is the dropdown that is shown when clicking on the unit (e.g. "px"):

<img width="190" alt="Screenshot 2022-09-13 at 12 15 12" src="https://user-images.githubusercontent.com/1083581/189876014-fd862d80-92ce-48d4-bb0f-d6362fda2bd1.png">
<img width="211" alt="Screenshot 2022-09-13 at 12 15 05" src="https://user-images.githubusercontent.com/1083581/189876035-cc87ce61-25f0-4d56-8124-d03661649e0a.png">

> + / – should show inside the control like the line height example.

The line height control doesn't show a unit, and so it would be using a `NumberControl`.

My question was about instances in which a `UnitControl` is used (which means that the unit dropdown is shown)

Reply:
Ah, I hear you know. Indeed, the idea is the + / – makes sense for unit-less controls. UnitControl should just work with the slider. I cannot think of a case where we'd have the + / – and unit right now.